### PR TITLE
chore: update runs-on value in python_release.yml

### PR DESCRIPTION
**Pull request by wjones127**
# Description

Latest Python release had a bunch of failures: https://github.com/delta-io/delta-rs/actions/runs/7095801050

Also doing some general cleanup.

TODO:

 * [x] Figure out why Linux job failed

# Related Issue(s)

# Documentation

### CODE REVIEW

File: .github/workflows/python_release.yml [hunk 1]
```diff
@@ -35,47 +35,23 @@ jobs:
...
       - uses: actions/checkout@v3
 
-      # We use extra recent Cargo.toml syntax, so we need at least Rust 1.71.0
-      - name: Install newer rust
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: default
-          toolchain: stable
-          override: true
-
```
Comment (wjones127) [inline .github/workflows/python_release.yml:49]: Newer runner has Rust 1.73.0: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#rust-tools

File: .github/workflows/python_release.yml [hunk 2]
```diff
@@ -35,47 +35,23 @@ jobs:
...
       - name: Publish to pypi (without sdist)
         uses: messense/maturin-action@v1
         env:
           MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
         with:
           target: ${{ matrix.target }}
           command: publish
-          args: -m python/Cargo.toml --no-sdist ${{ env.FEATURES_FLAG }}
-
-  release-pypi-mac-universal2:
```
Comment (wjones127) [inline .github/workflows/python_release.yml:59]: I don't think anyone really has need for this.

Comment (ion-elgreco) [reply .github/workflows/python_release.yml:59]: What are those feature flags setting there?

**Review by ion-elgreco — APPROVED** [review]

File: .github/workflows/python_release.yml (hunk 2 above)
Comment (wjones127) [reply .github/workflows/python_release.yml:59]: You mean this?

https://github.com/delta-io/delta-rs/blob/573cbd139a5607d72d12fc0480884d8a517275b3/.github/workflows/python_release.yml#L12-L14

**Review by ion-elgreco — APPROVED** [review]

**Pull request merged**